src/status_exception.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Exception with ops status"""
+
+import ops
+
+
+class StatusException(Exception):
```

Comment:
We can change it once we have an externalized upgrade lib. Otherwise, we have a hirerarchy of exception in the opensearch charm, and every exception should extend from the base `OpenSearchError` or a subclass - this allows to have default handling etc. We should stay consistent.